tests/Integration/Behaviour/Features/Context/CarrierFeatureContext.php

```diff
@@ -0,0 +1,375 @@
+     * @param $countryName
+     */
+    public function checkCountryWithNameExists($countryName)
+    {
+        if (!isset($this->countries[$countryName])) {
+            throw new \Exception('Country with name "' . $countryName . '" was not added in fixtures');
+        }
+    }
+
+    /**
+     * @Given /^There is a state with name (.+) and iso code (.+) in country named (.+) and zone named (.+)$/
+     */
+    public function setState($stateName, $stateIsoCode, $countryName, $zoneName)
+    {
+        $this->checkZoneWithNameExists($zoneName);
+        $this->checkCountryWithNameExists($countryName);
+        $state = new State();
+        $state->name = $stateName;
+        $state->iso_code = $stateIsoCode;
+        $state->id_zone = $this->zones[$zoneName]->id;
+        $state->id_country = $this->countries[$countryName]->id;
+        $state->add();
+        $this->states[$stateName] = $state;
+    }
+
+    /**
+     * @param $stateName
+     *
+     * @return State
+     */
+    public function getStateWithName($stateName)
+    {
+        return $this->states[$stateName];
+    }
+
+    /**
+     * @param $stateName
+     */
+    public function checkStateWithNameExists($stateName)
+    {
+        if (!isset($this->states[$stateName])) {
+            throw new \Exception('State with name "' . $stateName . '" was not added in fixtures');
+        }
+    }
+
+    /**
+     * @Given /^There is an address with name (.+) and post code (.+) in country named (.+) and state named (.+)$/
+     */
+    public function setAddress($addressName, $postCode, $countryName, $stateName)
+    {
+        $this->checkCountryWithNameExists($countryName);
+        $this->checkStateWithNameExists($stateName);
+        $address = new Address();
+        $address->id_country = $this->countries[$countryName]->id;
+        $address->id_state = $this->states[$stateName]->id;
+        $address->postcode = $postCode;
+        $address->lastname = 'lastname';
+        $address->firstname = 'firstname';
+        $address->address1 = 'address1';
+        $address->city = 'city';
+        $address->alias = 'alias';
+        $address->add();
+        $this->addresses[$addressName] = $address;
+    }
+
+    /**
+     * @Given /^Address with name (.+) is associated to customer with name (.+)$/
+     */
+    public function setAddressCustomer($addressName, $customerName)
+    {
+        $this->checkAddressWithNameExists($addressName);
+        $this->customerFeatureContext->checkCustomerWithNameExists($customerName);
+        $this->addresses[$addressName]->id_customer = $this->customerFeatureContext->getCustomerWithName($customerName)->id;
+        $this->addresses[$addressName]->update();
+    }
+
+    /**
+     * @param $addressName
+     */
+    public function checkAddressWithNameExists($addressName)
+    {
+        if (!isset($this->addresses[$addressName])) {
+            throw new \Exception('Address with name "' . $addressName . '" was not added in fixtures');
```

Comment:
Improvement to help debugging: print the list of known address names in the fixtures

Something like
```
            $knownAddresses = array_keys($this->addresses);
            throw new \Exception('Address with name "' . $addressName . '" was not added in fixtures, known names are '.$knownAddresses);
```

This is especially useful when it's just a typo 😉